Rename `data` variables to more readable name [Fixes #829]

EshaanAgg opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `alpha-x.x.x`: For stability testing: Only bug fixes accepted.
- `master`: Where the stable production ready code lies. Only security related bugs.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
Bugfix
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**
Fixes #829 

**Did you add tests for your changes?**
NA
<!--Yes or No. Note: Add unit tests or automation tests for your code.-->

**Snapshots/Videos:**
NA
<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**
NA
<!--Add link to Talawa-Docs.-->

**Summary**
Renamed the `data` variables to move understandable names wherever multiple of them being consumed in the same component. Also fixed the intermittent failing test in Roles.tsx.
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
No
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**
NA
<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes
<!--Yes or No-->

palisadoes commented:
It's not running the tests

EshaanAgg commented:
@anshgoyalevil Can you please take a look as to why the PR workflow isn't working?

EshaanAgg commented:
Yes @palisadoes I'm aware. Trying to debug the same.

palisadoes commented:
I think: 
1. You cannot have two `-run` statements one after the other, each has to have its own name
2. The `uses` at the top needs a `-name` too

anshgoyalevil commented:
@EshaanAgg 

Try using this Workflow https://gist.github.com/anshgoyalevil/5900c9742248e38a933de93f97ae7b57

I have indented the count line script. It is running.

palisadoes commented:
Add names for each of the steps it makes it easier for the person to see where the tests fail.

EshaanAgg commented:
@anshgoyalevil Thanks got it.
@palisadoes Have added the names for each job for easy debugging.

palisadoes commented:
@EshaanAgg 

I made the github actions changes. It should work now. Reopen the PR

anshgoyalevil commented:
@palisadoes 

It is still wrong. There shouldn't have ```- name``` and ```- run```

palisadoes commented:
> @palisadoes
> 
> It is still wrong. There shouldn't have `- name` and `- run`

Fixed

palisadoes commented:
It's working now

anshgoyalevil commented:
@palisadoes 

I guess minimum coverage should be 90 for now in ```min_coverage: 92.0``` line in ```pull-requests.yml``` file

palisadoes commented:
The last successfully merged PR was at 92%

- #824 

![image](https://user-images.githubusercontent.com/16875803/229331391-eeff18b6-e5dc-4a0d-a54d-f474533fd927.png)

EshaanAgg commented:
I agree with @anshgoyalevil. 92% is too strict a check right now, as even in the case of simple refactoring (as in this PR), the count of lines might increase (with the functionality remaining the same), and thus the test coverage might decrease.

palisadoes commented:
I'll reduce it to 91%

EshaanAgg commented:
Moreover, another problem the codebase still faces is the fact that most of the screens in the components use inline conditionals like `data ? data.length : 0`. 
The problem with test coverage in such an approach is that this line would always be covered by the tests, as both the conditions are written in the same line. Rewriting the same as 
```
data ?
  data.length : 
  0 
 ``` would then ensure that both the conditions are in different lines and thus enforce that both the lines are covered by tests.

palisadoes commented:
Please fix the conflicting files

ttaylor92 reviewed (changes requested):
Please resolve merge conflicts

EshaanAgg commented on src/screens/LoginPage/LoginPage.tsx:
```diff
@@ -140,7 +140,7 @@ function LoginPage(): JSX.Element {
       if (cPassword == signPassword) {
         try {
-          const { data } = await signup({
+          const { data: singUpData } = await signup({
```
Updating the variable is an option, but the term `data` is too general in my opinion, and leaves ambiguity as to what data we are referring to. Providing an alias gives the same more context, and defines the variable in only the scope that it would be consumed.

Rumone replied:
Yeah I agree, why not change the variable name instead?

Instead of
```...data: userData...```
Just use `userData`

EshaanAgg replied:
@Rumone It's cause we are using destructing here. By default, the `useQuery` hook returns the `data` object, which we can then alias to `userData`. If we were to access `userData` directly, the same would always be null as the same is never exported by the `useQuery` or analogous hooks.

EshaanAgg commented:
@palisadoes Fixed the conflicting files. The test is failing due to the improper tests written for the calender module, as highlighted in #911.

palisadoes commented:
The failing calendar tests should have been fixed. Please update the PR accordingly.

EshaanAgg commented:
@palisadoes Updated the PR. Can you please re-run the workflows? The formatting CI/CD seems to have crashed due to some error from GitHub's side.

EshaanAgg commented:
![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/96648934/9772f8c0-925d-4939-bb65-76922d28a467)
As you can see, sir, the code-coverage will increase itself after this PR is merged. 

The failing part is due to the coverage threshold on the repository, which in my opinion should be reduced as the total number of lines of code in the project has undergone considerable change since the last time the limit was set.
@palisadoes
